File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {

    int count=0;

    pair<int,int> postorder(TreeNode* root){

        if(!root){
            return {0,0};
        }

        pair<int,int> a = postorder(root->left);
        pair<int,int> b = postorder(root->right);
        pair<int,int> c = { (a.first+b.first+root->val) , (a.second+b.second+1)};
        if( ((c.first)/(c.second)) ==root->val ) count++;

        return c;

    }
public:
    int averageOfSubtree(TreeNode* root) {
        postorder(root);
        return count;
    }
};

File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int count=0;

    public int[] post(TreeNode root,int sum,int n){
        if(root==null){
            int arr[]={0,0};
            return arr;
        }

        int a[]=post(root.left,sum,n);
        int b[]=post(root.right,sum,n);

        int arr[]={a[0]+b[0]+root.val,a[1]+b[1]+1};
       

        if(root.val==(arr[0]/arr[1])){
            count++;
        }
        return arr;
    }
    public int averageOfSubtree(TreeNode root) {
        int sum=0;
        int n=0;
        post(root,sum,n);
        return count;
    }
}